### ISSUE TITLE
`gather` returns `Variable` with unknown size

### ISSUE THREAD

**Issue by baggepinnen** [malmaud/TensorFlow.jl#31]
I'm running into a problem with a tensor of unknown shape after having used `gather`
```julia
input_data = placeholder(Int32, shape=[20, 20])
embedding = get_variable("embedding", [10000, 200], Float32)
inputs = gather(embedding, input_data)
inputs = TensorFlow.split(1, 20, inputs)
julia> get_shape(inputs)
TensorShape[unknown]
```

I would like to send this into `nn.rnn` which produces the error
```
julia> outputs, state = nn.rnn(cell, inputs, initial_state=initial_state)
ERROR: Shape must be inferable
 in check_shape(::TensorFlow.ShapeInference.TensorShape) at /local/home/fredrikb/.julia/v0.5/TensorFlow/src/ops/rnn_cell.jl:23
 in (::TensorFlow.nn.rnn_cell.BasicRNNCell)(::TensorFlow.Tensor, ::TensorFlow.Tensor) at /local/home/fredrikb/.julia/v0.5/TensorFlow/src/ops/rnn_cell.jl:36
 in (::TensorFlow.nn.##12#13{TensorFlow.nn.rnn_cell.BasicRNNCell})() at /local/home/fredrikb/.julia/v0.5/TensorFlow/src/ops/nn.jl:88


I might be using this the wrong way, but as I understand it this works in python.

The context is https://github.com/baggepinnen/TensorFlow.jl/blob/ptbexample/examples/ptb_word_lm.jl#L165

**Comment by malmaud** [comment]
Someone just needs to implement shape inference for  `gather`. I'll probably get to it soon.

**Comment by baggepinnen** [comment]
I see, not an issue then, my apologies. 
I'll have a look at the inference and see if I can figure it out and contribute with a pull request.

**Comment by baggepinnen** [comment]
Closed with #32
